virttest/iscsi.py

```diff
@@ -833,5 +834,170 @@ def create_iSCSI(params, root_dir=data_dir.get_tmp_dir()):
                 )
             iscsi_instance = IscsiTGT(params, root_dir)
         else:
-            iscsi_instance = IscsiLIO(params, root_dir)
+            iscsi_instance = (
+                MultiTargetsIscsiLIO(params, root_dir)
+                if not params.get("target") and params.get("targets_luns")
+                else IscsiLIO(params, root_dir)
+            )
         return iscsi_instance
+
+
+class MultiTargetsIscsiLIO(IscsiLIO):
+    """
+    Iscsi class supporting multi-targets for the LIO backend used in RHEL7.
+    """
+
+    def __init__(self, params, root_dir):
+        """
+        Init the object based on the params and root_dir.
+        Note: 'targets_luns' in params is introduced at this class.
+              The structure of 'targets_luns':
+                targets_luns =
+                {
+                    ${target01}: {
+                        ${lun01} : {"emulated_image": ..., ...},
+                        ${lun02} : {"emulated_image": ..., ...},
+                    },
+                    ${target02}: {
+                        ${lun01} : {"emulated_image": ..., ...},
+                        ${lun02} : {"emulated_image": ..., ...},
+                    },
+                    ...
+                }
+              If there's only one lun without any special assigned requests,
+              lun should be set to "default_lun", for example:
+                targets_luns = { ${target01}: { "default_lun": {...}}}
+
+        :param params: parameters dict for iSCSI
+        :param root_dir: path for image
+        """
+        self._targets_luns = ast.literal_eval(params.get("targets_luns"))
+        # stores the iscsiLIO objects based on each targets
+        self._targets_mapping = {}
+        for target in self._targets_luns.keys():
```

Comment:
for target in self._targets_luns, pythonic way? please also check the following code below

Reply:
> for target in self._targets_luns, pythonic way? please also check the following code below

Like this way: ``for target in self._targets_luns:` ?

Reply:
@zhencliu Please take a look, thanks!